tests/Doctrine/Tests/DBAL/Functional/StatementTest.php

```diff
@@ -50,6 +54,75 @@ public function testStatementIsReusableAfterClosingCursor(): void
         self::assertEquals(2, $id);
     }
 
+    public function testFetchAllWithOneArgument(): void
+    {
+        $table = new Table('stmt_fetch_all');
+        $table->addColumn('a', 'integer');
+        $table->addColumn('b', 'integer');
+        $this->connection->getSchemaManager()->createTable($table);
+
+        $this->connection->insert('stmt_fetch_all', [
+            'a' => 1,
+            'b' => 2,
+        ]);
+
+        $stmt = $this->connection->prepare('SELECT a, b FROM stmt_fetch_all');
+        $stmt->execute();
+
+        self::assertEquals([[1, 2]], $stmt->fetchAll(FetchMode::NUMERIC));
+    }
+
+    /**
+     * @depends testFetchAllWithOneArgument
+     */
+    public function testFetchAllWithTwoArguments(): void
+    {
+        $driver = $this->connection->getDriver();
+
+        if ($driver instanceof IBMDB2Driver) {
```

Comment:
Should this explicitly allow only PDO-based drivers? They could be identified by the `PDOConnection`.

Reply:
Good idea. Let me know what you think of PDO Oracle and I'll refactor this.